src/driver/argument_parser.hpp

```diff
@@ -187,6 +187,13 @@ struct value_parser
     }
 };
 
+// version for std::optional object
+template <class T>
+struct value_parser<std::optional<T>>
+{
+    static T apply(const std::string& x) { return value_parser<T>::apply(x); }
```

Comment:
Actually this should return `std::optional<T>`: `static std::optional<T> apply(const std::string& x) { return value_parser<T>::apply(x); }`.